nixos/modules/programs/command-not-found/command-not-found.pl

```diff
@@ -23,6 +23,16 @@
 
 my $len = !defined $res ? 0 : scalar @$res;
 
+sub nix_shell_run {
+    my ($package, @ARGV) = @_;
+    if ($ENV{"NIX_AUTO_RUN_FLAKE"} // "") {
```

Comment:
We should add a nixos option for that.

Reply:
Are there nixos options for `NIX_AUTO_RUN` and `NIX_AUTO_RUN_INTERACTIVE`? I'm just setting environment variables atm: https://github.com/RyanGibb/nixos/blob/8fba04b5b45a4589f90094ae7843fd716ed11842/modules/common/default.nix#L71-L74

Reply:
We shouldn't add more (undocumented) environment variables, since how would a user even discover that they can set this? Ideally, we would use the `nix shell` code path automatically in a NixOS system built using flakes.